packages/wallet-manager/src/wallet-manager.ts

```diff
@@ -190,7 +190,7 @@ export class WalletManager extends EventEmitter {
     // Clean password from state
     this.#passphrase = '';
     // Emit event that wallet is locked
-    this.emit('lock');
+    await Promise.resolve(this.emit('lock'));
```

Comment:
@luizstacio Do you think we should have `async` stuff here in the future? It looks pretty `sync` to me; I'd guess the `async` notation was a trivial mistake like most things in this PR.